Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,324 @@
+                {
+                    // the requested qualityLevel index wasn't found, use highest available 
+                    PerformConsoleCommand(command, FixedValueString::format("%.*s/%d",
+                        AZ_STRING_ARG(visitArgs.m_jsonKeyPath), currentQualityLevel));
+                }
+            }
+            else
+            {
+                PerformConsoleCommand(command, visitArgs.m_jsonKeyPath);
+            }
+
+            return VisitResponse::Continue;
+        };
+        auto key = FixedValueString::format("%s/Settings", m_path.c_str());
+        AZ::SettingsRegistryVisitorUtils::VisitObject(*registry, callback, key);
+    }
+
+    AZ::PerformCommandResult QualityCVarGroup::PerformConsoleCommand(AZStd::string_view command, AZStd::string_view key)
+    {
+        using Type = AZ::SettingsRegistryInterface::Type;
+
+        auto registry = AZ::SettingsRegistry::Get();
+        auto console = AZ::Interface<AZ::IConsole>::Get();
+        if (!registry || !console)
+        {
+            return AZ::Failure("Missing required console or settings registry");
+        }
+
+        AZStd::string stringValue;
+        switch (registry->GetType(key))
+        {
+        case Type::FloatingPoint:
+            if (double value; registry->Get(value, key))
+            {
+                stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+            }
+            break;
+        case Type::Boolean:
+            if (bool value; registry->Get(value, key))
+            {
+                stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+            }
+            break;
+        case Type::Integer:
+            if (AZ::s64 value; registry->Get(value, key))
+            {
+                stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+            }
+            break;
+        case Type::String:
+        default:
+            {
+                registry->Get(stringValue, key);
+            }
+            break;
+        }
+
+        if (!stringValue.empty())
+        {
+            return console->PerformCommand(command, { stringValue }, AZ::ConsoleSilentMode::Silent);
+        }
+
+        return AZ::Failure(AZStd::string::format("Failed to stringify %.*s value, or it's empty.",
+            AZ_STRING_ARG(command)));
+    }
+
+    bool QualityCVarGroup::StringToValue(const AZ::ConsoleCommandContainer& arguments)
+    {
+        auto registry = AZ::SettingsRegistry::Get();
+        if (registry == nullptr)
+        {
+            return false;
+        }
+
+        if (!arguments.empty())
+        {
+            AZ::CVarFixedString value{ arguments.front() };
+
+            // GetQualityLevel will return QualityLevel::Invalid if unable
+            // to convert the provided value to a valid QualityLevel
+            QualityLevel qualityLevel = GetQualityLevel(value);
+
+            if (qualityLevel == QualityLevel::LevelFromDeviceRules)
+            {
+                // TODO use device rules

```

Comment:
Is this `//TODO` captured in a GHI anywhere?

Reply:
Not a GHI but it is part of the RFC referenced in this PR.